packages/firestore/test/unit/core/query.test.ts

```diff
@@ -852,6 +852,51 @@ describe('Query', () => {
     assertQueryMatches(query5, [doc3], [doc1, doc2, doc4, doc5]);
   });
 
+  it('matches composite queries with multiple inequality', () => {
+    const doc1 = doc('collection/1', 0, { a: 1, b: 0 });
+    const doc2 = doc('collection/2', 0, { a: 2, b: 1 });
+    const doc3 = doc('collection/3', 0, { a: 3, b: 2 });
+    const doc4 = doc('collection/4', 0, { a: 1, b: 3 });
+    const doc5 = doc('collection/5', 0, { a: 1, b: 1 });
+
+    // a>1 || b!=1.
+    const query1 = query(
+      'collection',
+      orFilter(filter('a', '>', 1), filter('b', '!=', 1))
+    );
+    assertQueryMatches(query1, [doc1, doc2, doc3, doc4], [doc5]);
+
+    // (a>=1 && b==0) || (a==1 && b!=1)
+    const query2 = query(
+      'collection',
+      orFilter(
+        andFilter(filter('a', '>=', 1), filter('b', '==', 0)),
+        andFilter(filter('a', '==', 1), filter('b', '!=', 1))
+      )
+    );
+    assertQueryMatches(query2, [doc1, doc4], [doc2, doc3, doc5]);
+
+    // a<=2 && (a!=1 || b<3).
+    const query3 = query(
+      'collection',
+      andFilter(
+        filter('a', '<=', 2),
+        orFilter(filter('a', '!=', 1), filter('b', '<', 3))
+      )
+    );
+    assertQueryMatches(query3, [doc1, doc2, doc5], [doc3, doc4]);
+
+    // (a!=1 || b!=1) && (a>=3 || b<3)
```

Comment:
this shouldn't be allowed, right? (multiple !=)

Reply:
Yeap, it shouldn't. This composite query didn't check the conflicting operators at all.